test/lightning/projects/merge_projects_test.exs

```diff
@@ -2311,4 +2340,431 @@ defmodule Lightning.Projects.MergeProjectsTest do
+
+      insert(:workflow_version,
+        workflow: source_workflow,
+        hash: "abc123",
+        source: "app"
+      )
+
+      insert(:workflow_version,
+        workflow: target_workflow,
+        hash: "def456",
+        source: "app"
+      )
+
+      assert MergeProjects.has_diverged?(source_project, target_project)
+    end
+
+    test "returns false when target has no workflows" do
+      target_project = insert(:project)
+
+      source_project = insert(:project, parent_id: target_project.id)
+      source_workflow = insert(:workflow, project: source_project)
+
+      insert(:workflow_version,
+        workflow: source_workflow,
+        hash: "abc123",
+        source: "app"
+      )
+
+      refute MergeProjects.has_diverged?(source_project, target_project)
+    end
+
+    test "returns true when any workflow in target has different version" do
+      target_project = insert(:project)
+
+      target_workflow_1 =
+        insert(:workflow, project: target_project, name: "Workflow 1")
+
+      target_workflow_2 =
+        insert(:workflow, project: target_project, name: "Workflow 2")
+
+      insert(:workflow_version,
+        workflow: target_workflow_1,
+        hash: "abc123",
+        source: "app"
+      )
+
+      insert(:workflow_version,
+        workflow: target_workflow_2,
+        hash: "def456",
+        source: "app"
+      )
+
+      source_project = insert(:project, parent_id: target_project.id)
+
+      source_workflow_1 =
+        insert(:workflow, project: source_project, name: "Workflow 1")
+
+      source_workflow_2 =
+        insert(:workflow, project: source_project, name: "Workflow 2")
+
+      insert(:workflow_version,
+        workflow: source_workflow_1,
+        hash: "abc123",
+        source: "app"
+      )
+
+      insert(:workflow_version,
+        workflow: source_workflow_2,
+        hash: "def456",
+        source: "app"
+      )
+
+      insert(:workflow_version,
+        workflow: target_workflow_2,
+        hash: "xyz789",
+        source: "app"
+      )
+
+      assert MergeProjects.has_diverged?(source_project, target_project)
+    end
+
+    test "uses most recent version hash when multiple versions exist" do
+      target_project = insert(:project)
+
+      target_workflow =
+        insert(:workflow, project: target_project, name: "Workflow 1")
+
+      insert(:workflow_version,
+        workflow: target_workflow,
+        hash: "old_version",
+        source: "app",
+        inserted_at: DateTime.add(DateTime.utc_now(), -2, :hour)
+      )
+
+      insert(:workflow_version,
+        workflow: target_workflow,
+        hash: "latest_version",
+        source: "app",
+        inserted_at: DateTime.utc_now()
+      )
+
+      source_project = insert(:project, parent_id: target_project.id)
+
+      source_workflow =
+        insert(:workflow, project: source_project, name: "Workflow 1")
+
+      insert(:workflow_version,
```

Comment:
@elias-ba This feels like it may be a duplication of the  'returns true when target workflow version differs from sandbox version' - would suggest having multiple versions for the source here as well.